src/utils/viewer/viewer.cpp

```diff
@@ -105,6 +115,11 @@ class Viewer : public Magnum::Platform::Application {
   bool drawObjectBBs = false;
 
   Magnum::Timeline timeline_;
+
+  Cr::Containers::Pointer<Mn::Text::Renderer2D> performanceText_;
```

Comment:
Maybe worth to factor out Text drawing in separate tiny class that will have `init`, `draw`, `update` to keep viewer clean and make code re-usable for time when we replace `viewer` with simulator usage.

Reply:
I second this.  Will make it much easier in the long run.

Reply:
This is the 1st time to introduce it to the system, and it is non-trivial (as you can see it in this PR.)
I will have to make it work first.
The further improvement will be in the future.

Reply:
I get doing things like this to make it work, but I don't agree with not fixing it up once it's working.  The "and it is non-trivial (as you can see it in this PR.)" further reinforces that it should be compartmentalized so when someone decided to refactor the viewer (and it's definitely overdue for one), they don't have to go through what you went through to get it working again.

Reply:
@erikwijmans :
I appreciate your comments, which makes a lot of sense. Please do not get me wrong. I certainly pay attention to the quality of the code.

- Code to display text on GUI is even more non-trivial if written with raw OpenGL from scratch. Magnum library already encapsulates and hides a large amount of tedious details, and provided high quality APIs to be used here. So the code (implemented in this PR) may  not be that ugly to start with, first of all.

- "not fixing it up once it's working." I never say or imply that I would not "fix" it. Actually "improve" would be a more proper word here, since there is nothing wrong with the current implementation. I said "further improvement is in the future". It may just not be in this PR. 

- Why not make the improvement in this PR? I need to clarify that I am quite open to this suggestion. But it will slow me down. This is the 1st PR in the "rendering acceleration" project, and it is the simplest one. My plan was to make a quick PR (to be committed 1-2 days) and put my focus on the following challenging work, which needs much more time. I have a lot of experience on this topic and I know which part is the most difficult one. So I have to prioritize the work, and invest my time wisely given the limited time budget (to hit the Q4 goal).

- I cared about the quality of the code. Actually because of this reason, I was thinking to postpone the improvement because it needs a careful design, and it takes time. I indeed thought about it last week. The core idea of the improvement is to create a higher level, helper class to further encapsulate and drive the fontManager, font, cache, Renderer2D etc. Two ways to do it. One is to do it in the Magnum library, and the other is to do it in our libraries. (I was planning to hear the opinions from Vladimir first). Either way relates to a lot of detailed design decisions, user APIs, location of the fonts etc.

Again, I am willing to do the improvement here, if eventually we agree it must be done in this PR. The above bulletin points are simply to let you understand a bit more about my situation and thoughts.

Thanks.

Reply:
Eh. What about this, then -- I'll briefly pause my work on texture compression / asset import and create some `DebugTools::FpsCounter` thing that wraps this up into an "oneliner" with a similar convenience/flexibility tradeoffs like [DebugTools::screenshot()](https://doc.magnum.graphics/magnum/namespaceMagnum_1_1DebugTools.html#a66cd81ac668af7f8dffcd27ee47b2019) we're already using in the viewer.

That means all work put into this so far will be thrown away, but if that saves you from spending even more time on it, why not. And I was planning to implement such a thing anyway, only you needed it sooner than I could have done it.

Reply:
@mosra : I appreciate the help you offered. Thank you. 
Actually I knew you are busy with the texture staff, so I started it by myself.

What I really need is a generic "text display" class, much more than a "FpsCounter". Besides FPS, it will also help me to display e.g., how many drawables or triangles are culled later. I think such an advanced class (a wrapper) will be built on top of the fontManager, font, cache, projMtx etc (as I mentioned in previous comments) to simplify the operations here. Like I say one proper way is to do it within magnum, and the other way is to do it in our lib. Does it make sense to you? 

Let me know if you are going to do it or let me do it. I am happy to.

Reply:
@mosra: I am going to code it up. You can focus on your current work. :-)

Reply:
It would be first just for FPS, later extended with info coming from [ARB_pipeline_statistics_query](https://www.khronos.org/registry/OpenGL/extensions/ARB/ARB_pipeline_statistics_query.txt) etc., possibly plotting a graph of the values over time. Once I'm done with the asset pipeline rework, I'll need it "for marketing purposes" to monitor resource use before/after :)